qa/integration-tests/src/test/java/io/camunda/zeebe/it/util/GrpcClientRule.java

```diff
@@ -104,12 +105,21 @@ public ZeebeClient getClient() {
   }
 
   public void waitUntilDeploymentIsDone(final long key) {
-    waitUntil(
-        () ->
-            RecordingExporter.commandDistributionRecords()
-                .withIntent(CommandDistributionIntent.FINISHED)
-                .withRecordKey(key)
-                .exists());
+    if (getPartitions().size() > 1) {
+      waitUntil(
+          () ->
+              RecordingExporter.commandDistributionRecords()
+                  .withIntent(CommandDistributionIntent.FINISHED)
+                  .withRecordKey(key)
+                  .exists());
+    } else {
+      waitUntil(
+          () ->
+              RecordingExporter.deploymentRecords()
+                  .withIntent(DeploymentIntent.CREATED)
+                  .withRecordKey(key)
+                  .exists());
+    }
```

Comment:
🔧 We could instead use the RecordingExporter to find the DeploymentIntent.CREATED on all partitions and then check that it exists on all partitions. This is slightly more complex but semantically better fitting. Alternatively, we could check CREATED on 1 partition, and check that the DISTRIBUTION of that key **and** intent was FINISHED

Reply:
> We could instead use the RecordingExporter to find the DeploymentIntent.CREATED on all partitions and then check that it exists on all partitions

You mean something like this?

```java
    for (final Integer partition : getPartitions()) {
      waitUntil(
          () ->
              RecordingExporter.deploymentRecords()
                  .withIntent(DeploymentIntent.CREATED)
                  .withPartitionId(partition)
                  .withRecordKey(key)
                  .exists());
    }
```

Reply:
No, that wouldn't really scale well. We can do it one check:

```java
Assertions.assertThat(
  RecordingExporter.deploymentRecords()
    .withIntent(DeploymentIntent.CREATED)
    .withRecordKey(key)
    .limit(getPartitions().size())
).hasPartitionIds(getPartitions()) 
```